src/main/java/org/mondemand/Client.java

```diff
@@ -1270,13 +1297,17 @@ private synchronized void dispatchContextStats()
       }
 
       Context[] contexts = newContexts.toArray(new Context[0]);
-
-      for (Transport t : transports) {
-        try {
-          t.send(programId, statsMsgs.toArray(new StatsMessage[0]), null,
-                 contexts);
-        } catch(TransportException te) {
-          errorHandler.handleError("Error calling Transport.sendStats()", te);
+      List<Transport> transportList = transports.get(EventType.STATS);
+
+      if (transportList != null) {
+        for (Transport t : transportList) {
```

Comment:
why not:
for (Transport t: transports.get(EventType.STATS))
and get rid of the above if statement.